masschallenge/app-allocator-scripts
# [AC-5574] Abstract Feature concept for generator and allocator

To test:
- Make sure that the old way of running allocate.py and generate.py still work.
- Test generate.py:
-- Copy distributions.csv (e.g., `cp distributions.csv test-dists.csv`) and make some noticeable changes such as changing the relative weights of male/female.
-- Run generate.py specifying the new CSV as the final command line arg (e.g., `python3 generate.py 10 100 test-dists.csv > test-scenario.csv`) and then check the results.
-- Try some other changes to the distributions input file including add more features and see what happens.

- Test allocate.py
-- Copy criteria.csv (e.g., `cp criteria.csv test-crit.csv`) and make some noticeable changes in the criteria like removing everything except the reads criteria.
-- Run allocate.py specifying the new CSV as the final command line arg (e.g., `python3 allocate.py example.csv ordered_queues test-crit.csv`) and then check that the results are different as expected based on the change.
-- Try some other changes to the criteria and with different CSVs created by generate.py.  The set of features should be completely flexible depending what you give in the distributions CSV.

Note: I did not attempt refactoring analyze.py.  This is going to be a lot of work so I created a new ticket for it and we can debate whether it's worth doing.
